### PR TITLE
fix(AAE-11533): add wait for chart before propagate downstream

### PR DESCRIPTION
**Pull request by igdianov**
- [x] add wait for chart before propagating downstream
- [x] refactor publish job to reuse actions from alfresco-build-tools

### CODE REVIEW

File: .github/workflows/rc.yml [hunk 1]
```diff
@@ -32,7 +33,13 @@ jobs:
           helm-repository-base-url: ${{ env.HELM_REPO_BASE_URL }}
           helm-repository-token: ${{ secrets.BOT_GITHUB_TOKEN }}
           git-username: ${{ secrets.BOT_GITHUB_USERNAME }}
-          do-push: ${{ github.event_name == 'push' }}
```
Comment (atchertchian) [inline .github/workflows/rc.yml:35]: did you remove this on purpose?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -75,6 +90,12 @@ jobs:
       VERSION: ${{ needs.publish.outputs.version }}
       DEVELOPMENT_BRANCH: ${{ github.ref_name }}
     steps:
+      - uses: Activiti/activiti-scripts/.github/actions/wait-for-chart@master
```
Comment (atchertchian) [inline .github/workflows/main.yml:93]: I don't really follow what we're waiting for: for the helm chart that we just published to be available for use?

File: .github/workflows/rc.yml (hunk 1 above)
Comment (igdianov) [reply .github/workflows/rc.yml:35]: Yes, we don't need to have this condition for push on tag

Comment (atchertchian) [reply .github/workflows/rc.yml:35]: ah because we're in the rc workflow, ok

File: .github/workflows/main.yml (hunk 2 above)
Comment (igdianov) [reply .github/workflows/main.yml:93]: The Helm chart is published to gh-pages branch, which triggers Github action publish workflow to update Github site. Without wait, there will be problems when promoting new versions downstream trying to lint, build and package new version

Comment (atchertchian) [reply .github/workflows/main.yml:93]: so we're waiting, potentially for more than 4 minutes, in a running build, for another CI process to go through?

Comment (igdianov) [reply .github/workflows/main.yml:93]: Yes, it will not interfere with another CI build on RC tag push

Comment (erdemedeiros) [reply .github/workflows/main.yml:93]: The builds for [github pages](https://github.com/Activiti/activiti-cloud-helm-charts/actions) are taking about 2 minutes to run. I don't think we can do much about this (unless migrating from gh pages to something else: same approach used for [Alfresco/charts](https://github.com/Alfresco/charts)? But this is out of of the scope of this epic).

**Review by erdemedeiros — APPROVED** [review]
LGTM!

**Pull request merged**